bokehjs/src/coffee/common/selector.coffee

```diff
@@ -10,8 +10,10 @@ class Selector extends HasProps
     @set('timestamp', new Date(), {silent: silent})
     @set('final', final, {silent: silent})
     if append
+      if @get('indices').get_model()
+        indices.get_model = @get('indices').get_model
+        indices.get_view = @get('indices').get_view
       indices['0d'].indices =  _.union(@get('indices')['0d'].indices, indices['0d'].indices)
-      indices['0d'].glyph =  @get('indices')['0d'].glyph or indices['0d'].glyph
       indices['1d'].indices =  _.union(@get('indices')['1d'].indices, indices['1d'].indices)
       indices['2d'].indices =  _.union(@get('indices')['2d'].indices, indices['2d'].indices)
```

Comment:
I spoke in haste. This combines selections on a single *data source*, which might be driving several glyphs. The dimensionality reflects this: selection on "scatters" are just a 1d list of indices. What about selections on multilines? Well, you have a 1d list of indices that could represent that the k'th line has been "hit" *at all*, but you might want more information, i.e. which of the individual points of the k'th were specifically hit? "0d" is for a single line, where you just want to know if the line has been hit at all, this tells you that, and then "glyph" will tell you which glyph was actually the one the mouse hit... honestly all this should be re-thought, but data sources should be simplified at the same time, so not sure when that can happen.